python/cudf/cudf/core/dataframe.py

```diff
@@ -3528,66 +3528,144 @@ def quantile(
     #
     # Stats
     #
+    def _as_common_dtype(self):
+        common_dtype = np.find_common_type(self.dtypes, [])
+        coerced = self._apply_support_method("astype", dtype=common_dtype)
+        return coerced
+
+    def _prepare_for_rowwise_op(self):
+        """Prepare a DataFrame for CuPy-based row-wise operations.
+
+        Returns
+        -------
+        CuPy.NDArray
+        """
+        if not utils.IS_CUPY_AVAILABLE:
+            msg = (
+                "Row-wise operations currently require CuPy. "
+                " Please install CuPy to use these operations."
+            )
+            raise ImportError(msg)
+
+        import cupy as cp
+
+        if any([col.has_null_mask for col in self._columns]):
+            msg = (
+                "Row-wise operations do not currently support columns with "
+                "null values. Consider removing them with .dropna() "
+                "or using .fillna()."
+            )
+            raise ValueError(msg)
+
+        coerced = self._as_common_dtype()
+        arr = cp.asarray(coerced.as_gpu_matrix())
```

Comment:
This will return a column-major matrix, where you likely want row-major.
```suggestion
        arr = cp.asarray(coerced.as_gpu_matrix(order='C'))
```